src/gpuarray/buffer.h

```diff
@@ -24,25 +24,31 @@ extern "C" {
 #endif
 struct _gpudata;
 
+/**
+ * Opaque struct for buffer data.
+ */
+typedef struct _gpudata gpudata;
+
 struct _gpucontext;
 
 /**
  * Opaque struct for context data.
  */
 typedef struct _gpucontext gpucontext;
 
+struct _gpukernel;
 
 /**
- * Opaque struct for buffer data.
+ * Opaque struct for kernel data.
  */
-typedef struct _gpudata gpudata;
+typedef struct _gpukernel gpukernel;
```

Comment:
It does not change anything in functionality. I reordered them in order to be in line :P